swaywm/wlroots
# xwm clipboard sync, 2nd edition

- [x] MIME type from xwayland to wayland
- [x] MIME type from wayland to xwayland
- [x] Fix `xclip`
- [x] Don't send selection to clients that don't have focus

Weston doesn't implement selection MIME types, so we can't use it as a starting point.

Test plan:
* Copy text in wayland, paste it in xwayland
* Copy text in xwayland, paste it in wayland
* Copy text in xwayland, paste it in xwayland
* Same with rich text
* Copy something, run `xclip -o -t TARGETS -selection clipboard`, check MIME types
* Copy something, run `xclip -o -selection clipboard`
* Run `echo "poi" | xclip -i -selection clipboard`, paste somewhere

Test clients:
* Wayland: gedit, epiphany, lowriter
* Xwayland: gedit, chromium, lowriter

Updates #295
